code/src/hooks/use-on-click-outside/useOnClickOutside.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { useRef } from "react";
+
+import { useOnClickOutside } from "./useOnClickOutside";
+
+const TestComponent = ({
+  onClickOutside
+}: {
+  onClickOutside: (event: MouseEvent | TouchEvent) => void;
+}) => {
+  const ref = useRef<HTMLDivElement>(null);
+  useOnClickOutside(ref, onClickOutside);
+
+  return (
+    <div>
+      <div ref={ref} data-testid="inside">
+        Inside
+      </div>
+      <div data-testid="outside">Outside</div>
+    </div>
+  );
+};
+
+let removeEventListenerSpy: jest.SpyInstance;
+let handleClickOutside: jest.Mock;
+
+const setup = () => {
+  handleClickOutside = jest.fn();
+  const utils = render(<TestComponent onClickOutside={handleClickOutside} />);
+  return { ...utils };
```

Comment:
Why you did it? Just return `render(<TestComponent onClickOutside={handleClickOutside} />` from the function. Also we should prefer using `renderWithProviders` if possible

Reply:
For this case I need unmount for this component, so I decided to use common render. Thank you fixed